### PR TITLE
[CARBONDATA-1098] change PageStatistics to use exact type

### PR DESCRIPTION
**Pull request by jackylk**
Currently page statistics uses Object internally, which requires boxing/unboxing when processing it. 
This PR changes it to use exact type as schema defines

### CODE REVIEW

**Comment by jackylk** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2454/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/335/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2465/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/349/

File: core/src/main/java/org/apache/carbondata/core/datastore/page/statistics/ColumnPageStatsVO.java [hunk 1]
```diff
@@ -195,15 +192,36 @@ private int getDecimalCount(double value) {
...
   }
 
   public Object getMax() {
-    return max;
-  }
-
-  public Object nonExistValue() {
-    return nonExistValue;
+    switch (dataType) {
+      case SHORT:
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/datastore/page/statistics/ColumnPageStatsVO.java]: please use the same datatype set in all switch case statements of this class.

**Pull request closed without merging**